### PR TITLE
Feat: Add warmup_ratio

### PR DESCRIPTION
**Pull request by NanoCode012**
Untested!

It is a hassle to do warmup_steps without knowing full steps.

If anyone is available, feel free to give this a try.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -676,6 +676,7 @@ micro_batch_size: 2
 eval_batch_size:
 num_epochs: 4
 warmup_steps: 100
+warmup_ratio: 0.05
```
Comment (winglian) [inline README.md]: Should probably remove warmup steps.line before this one then

**Review by winglian — APPROVED** [review]
Good to go once the example yaml is updated

File: README.md (hunk 1 above)
Comment (NanoCode012) [reply README.md]: Hm, would it be better if I add a comment saying that they're exclusive? At the very least, we want to document this config.

Comment (NanoCode012) [reply README.md]: or did you mean, I should also update all the example yaml?

Comment (NanoCode012) [reply README.md]: I have updated readme to say they're exclusive

**Comment by NanoCode012** [comment]
Tested that it runs.
